src/analysis/processing/qgsalgorithmtinmeshcreation.cpp

```diff
@@ -0,0 +1,180 @@
+{
+  return QObject::tr( "Mesh" );
+}
+
+QString QgsTinMeshCreationAlgorithm::groupId() const
+{
+  return QStringLiteral( "mesh" );
+}
+
+QString QgsTinMeshCreationAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "TIN mesh creation from vector layers" );
+}
+
+QString QgsTinMeshCreationAlgorithm::name() const
+{
+  return QStringLiteral( "tinmeshcreation" );
+}
+
+QString QgsTinMeshCreationAlgorithm::displayName() const
+{
+  return QObject::tr( "TIN Mesh Creation" );
+}
+
+QgsProcessingAlgorithm *QgsTinMeshCreationAlgorithm::createInstance() const
+{
+  return new QgsTinMeshCreationAlgorithm();
+}
+
+void QgsTinMeshCreationAlgorithm::initAlgorithm( const QVariantMap &configuration )
+{
+  Q_UNUSED( configuration );
+  addParameter( new QgsProcessingParameterTinInputLayers( QStringLiteral( "SOURCE_DATA" ), QObject::tr( "Input layers" ) ) );
+
+  const QList<QgsMeshDriverMetadata> &driverList =
+    QgsProviderRegistry::instance()->providerMetadata( QStringLiteral( "mdal" ) )->meshDriversMetadata();
+
+  for ( const QgsMeshDriverMetadata &driverMeta : driverList )
+    if ( driverMeta.capabilities() & QgsMeshDriverMetadata::CanWriteMeshData )
+      mAvailableFormat.append( driverMeta.name() );
+
+  addParameter( new QgsProcessingParameterEnum( QStringLiteral( "MESH_FORMAT" ), QObject::tr( "Output format" ), mAvailableFormat, false, 0 ) );
+  addParameter( new QgsProcessingParameterCrs( QStringLiteral( "CRS_OUTPUT" ), QObject::tr( "Output Coordinate System" ), QVariant(), true ) );
+  addParameter( new QgsProcessingParameterFileDestination( QStringLiteral( "OUTPUT_MESH" ), QObject::tr( "Output File" ) ) );
+}
+
+bool QgsTinMeshCreationAlgorithm::prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  Q_UNUSED( feedback );
+  const QVariant layersVariant = parameters.value( parameterDefinition( QStringLiteral( "SOURCE_DATA" ) )->name() );
+  if ( layersVariant.type() != QVariant::List )
+    return false;
+
+  const QVariantList layersList = layersVariant.toList();
+
+  QgsCoordinateReferenceSystem destinationCrs = parameterAsCrs( parameters, QStringLiteral( "CRS_OUTPUT" ), context );
+  if ( !destinationCrs.isValid() )
+    destinationCrs = context.project()->crs();
+
+  for ( const QVariant &layer : layersList )
+  {
+    if ( feedback && feedback->isCanceled() )
+      return false;
+
+    if ( layer.type() != QVariant::Map )
+      continue;
+    const QVariantMap layerMap = layer.toMap();
+    const QString layerId = layerMap.value( QStringLiteral( "Id" ) ).toString();
+    const QgsProcessingParameterTinInputLayers::Type type =
+      static_cast<QgsProcessingParameterTinInputLayers::Type>( layerMap.value( QStringLiteral( "Type" ) ).toInt() );
+    int attributeIndex = layerMap.value( QStringLiteral( "AttributeIndex" ) ).toInt();
+
+    QgsVectorLayer *vectorLayer = context.project()->mapLayer<QgsVectorLayer *>( layerId );
```

Comment:
This part should be reworked so that we aren't taking layers from the project based on layer id. Rather, the source map created by QgsProcessingParameterTinInputLayers shouldn't have the id key and should instead have a "source" string value, which can be any valid processing feature source identifier. Then this loop would call `parameterAsSource` in order to retrieve the source ready for the call the getFeatures() on it.